### ISSUE TITLE
A template admin can not access the Web Terminal of another user's workspace

### ISSUE THREAD

**Issue by matifali** [coder/coder#9082]
![image](https://github.com/coder/coder/assets/10648092/282b7980-e39b-40fc-9e6f-f70c398c7492)

**Comment by sharkymark** [comment]
Were they supposed to be able to? Owner roles can unless disabled.

**Comment by matifali** [comment]
@sharkymark My suggestion is that if this is intended behavior, we should hide the button or change the error message to say `You do not have access.`

Also if a template admin can not access any functionality of the other user's workspaces, we should remove showing them all workspaces.

**Comment by MrPeacockNLB** [comment]
If it is indented to then we need a switch to disable this. Opening and working with someone else workspace breaks security rules as one (a template admin) can do things in other names!

A Workspace can contain OIDC token and GitHub auth token.

**Comment by matifali** [comment]
Let me rephrase it. Hide and disallow other users workspaces for template admins. They already can't access the terminal and coder_apps. So there is no point in listing them and allow them to navigate to the workspace page of another user.

**Comment by matifali** [comment]
@bpmct for comments

**Comment by bpmct** [comment]
> Let me rephrase it. Hide and disallow other users workspaces for template admins. They already can't access the terminal and coder_apps. So there is no point in listing them and allow them to navigate to the workspace page of another user.

Yeah I agree

**Comment by bpmct** [comment]
Let's close these and make a meta issue that coders this and #9081

**Comment by bpmct** [comment]
Closing. We address this in #9175
